File: RedeDoDINF.py
# 2522
from sys import stdin, stdout

def distance(x1, x2, y1, y2):
    return ((x2-x1)**2 + (y2-y1)**2)**0.5

while True:
    try:
        num_computers = int(stdin.readline())

        adj = []

        coord = []

        for i in range(num_computers):
            coord.append([int(x) for x in stdin.readline().split(" ")])

        # É calculado apenas a diagonal inferior da matriz para poupar processamento   
        for i in range(len(coord)):
            row = []
            for j in range(i):
                row.append(distance(coord[i][0], coord[j][0], coord[i][1], coord[j][1]))
            adj.append(row)

        connected_nodes = [0]
        not_connected_nodes = list(range(1, num_computers))
        total_length = 0

        while len(connected_nodes) < num_computers:
            smallest_distance = 999999
            closest_node = None

            for node in connected_nodes:
                for i in not_connected_nodes:
                    n = i
                    if node < i:
                        i,node = node,i

                    if adj[node][i] < smallest_distance:
                        smallest_distance = adj[node][i]
                        closest_node = n
                        
            not_connected_nodes.remove(closest_node)
            connected_nodes.append(closest_node)
            total_length += smallest_distance

        stdout.write('{:.2f}\n'.format(round(total_length, 2)))
    except EOFError:
        exit()

File: RedeDoDINF.js
// Wrong awnser
// const input = require("fs").readFileSync("/dev/stdin", "utf8");
const input = require("fs").readFileSync("./input.txt", "utf8");
const lines = input.split("\n");

let nextLine = 0;
const range = (n) => [...Array(n).keys()];

while (lines[nextLine]) {
  const numComputers = Number(lines[nextLine++]);
  const adj = [];

  const distance = (x1, x2, y1, y2) =>
    Math.sqrt(Math.pow(x2 - x1, 2) + Math.pow(y2 - y1, 2));
  const coord = [];

  range(numComputers).forEach(() => {
    coord.push(lines[nextLine++].split(" ").map(Number));
  });

  range(numComputers).forEach((i) => {
    const row = [];
    for (let j in range(i)) {
      row.push(distance(coord[i][0], coord[j][0], coord[i][1], coord[j][1]));
    }
    adj.push(row);
  });

  const connectedNodes = [0];
  let notConnectedNodes = [...range(numComputers)];
  notConnectedNodes.shift();
  let totalLength = 0;

  while (connectedNodes.length < numComputers) {
    let smallestDistance = 999999;
    let closestNode;

    connectedNodes.forEach((node) => {
      notConnectedNodes.forEach((possibleNode) => {
        const bkpPossibleNode = possibleNode;
        if (node < possibleNode) {
          const aux = node;
          node = possibleNode;
          possibleNode = aux;
        }

        if (adj[node][possibleNode] < smallestDistance) {
          smallestDistance = adj[node][possibleNode];
          closestNode = bkpPossibleNode;
        }
      });
    });

    notConnectedNodes = notConnectedNodes.filter(
      (node) => node !== closestNode
    );
    connectedNodes.push(closestNode);
    totalLength += smallestDistance;
  }

  console.log(totalLength.toFixed(2));
}
